fab3/filters.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package fab3
+
+import (
+	"encoding/hex"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/hyperledger/fabric-chaincode-evm/fab3/types"
+)
+
+type filterEntry interface {
+	LastAccessTime() time.Time
+	Filter(s *ethService) ([]interface{}, error)
+}
+
+type logsFilter struct {
+	logArgs         *types.GetLogsArgs
+	latestBlockSeen uint64
+	lastAccessTime  time.Time
+}
+
+func (f *logsFilter) LastAccessTime() time.Time {
+	return f.lastAccessTime
+}
+
+func (f *logsFilter) Filter(s *ethService) ([]interface{}, error) {
+	s.logger.Debug("lastblockseen before filtering", f.latestBlockSeen)
+	currentLatestBlockSeen, err := s.parseBlockNum("latest")
+	if err != nil {
+		return nil, err
+	}
+	s.logger.Debug("latest blockseen", currentLatestBlockSeen)
+
+	var l []interface{}
+	if f.latestBlockSeen < currentLatestBlockSeen {
+		var logs []types.Log
+		err = s.GetLogs(nil, f.logArgs, &logs)
+		if err != nil {
+			return nil, errors.Wrap(err, "GetLogs call failed")
+		}
+		l = make([]interface{}, 0, len(logs))
+		for _, v := range logs {
+			l = append(l, v)
+		}
+	}
+	// update the filter
+	f.lastAccessTime = time.Now()
+	f.logArgs.FromBlock = strconv.FormatUint(currentLatestBlockSeen+1, 16)
```

Comment:
We shouldn't edit the FromBlock as this will affect the filter when using `eth_getFilterLogs`

Reply:
I think we should be updating latestBlockSeen to currentLatestBlockSeen here instead of updating the From Block.

Reply:
Yea, I'm not sure what I was thinking here. Very weird looking to me now.